File: Backend_IPCA_Gym/LayerBOL/Models/Cliente.cs
namespace LayerBOL.Models
{
    public class Cliente
    {
        /// <summary>
        /// Id do cliente
        /// </summary>
        /// <example>1</example>
        public int id_cliente { get; set; }
        /// <summary>
        /// Id do ginásio em que o cliente está inscrito
        /// </summary>
        /// <example>1</example>
        public int id_ginasio { get; set; }
        /// <summary>
        /// id do plano nutricional selecionado pelo cliente, podendo este ser nulo
        /// </summary>
        /// <example>1</example>
        public int? id_plano_nutricional { get; set; }
        /// <summary>
        /// Nome do cliente
        /// </summary>
        /// <example>Joaquim Botelho</example>
        public string nome { get; set; }
        /// <summary>
        /// Mail do cliente
        /// </summary>
        /// <example>jjj@gmail.com</example>
        public string mail { get; set; }
        /// <summary>
        /// Contacto do cliente
        /// </summary>
        /// <example>a21111@gmail.com</example>
        public int telemovel { get; set; }
        /// <summary>
        /// Password do cliente
        /// </summary>
        /// <example>aaa222</example>
        public string pass_salt { get; set; }
        /// <summary>
        /// Representação criptografica da password do cliente
        /// </summary>
        /// <example>grtbgvr5j945-tj48fv94yugb4845</example>
        public string? pass_hash { get; set; }
        /// <summary>
        /// Peso do cliente em kg sendo este inicializado nulo
        /// </summary>
        /// <example>65.7</example>
        public double? peso { get; set; }
        /// <summary>
        /// Altura do cliente em cm sendo este inicializado nulo
        /// </summary>
        /// <example>176</example>
        public int? altura { get; set; }
        /// <summary>
        /// Gordura percentual do cliente sendo este inicializado nulo
        /// </summary>
        /// <example>10.5</example>
        public double? gordura { get; set; }
        /// <summary>
        /// Foto de perfil do cliente podendo ser nulo
        /// </summary>
        /// <example>C:\OneDrive\joaquim.png</example>
        public string? foto_perfil { get; set; }
        /// <summary>
        /// Estado do cliente (Ativo ou Inativo)
        /// </summary>
        /// <example>Inativo</example>
        public string estado { get; set; }

    }
}

File: IpcaGym2/app/src/main/java/vough/example/ipcagym/data_classes/Cliente.kt
package vough.example.ipcagym.data_classes

import org.json.JSONObject
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/*data class Cliente(
    val id_cliente: Int,
    val id_ginasio: Int,
    val id_plano_nutricional: Int,
    val nome: String,
    val mail: String,
    val telemovel: String,
    val pass_salt: String,
    val pass_hash: String,
    val peso: Double,
    val altura: Int,
    val gordura: Double,
    val foto_perfil: String,
    val estado: String
)*/

class Cliente{
    var id_cliente: Int? = null
    var id_ginasio: Int? = null
    var id_plano_nutricional: Int? = null
    var nome: String? = null
    var mail: String? = null
    var telemovel: Int? = null
    var pass_salt: String? = null
    var pass_hash: String? = null
    var peso: Double? = null
    var altura: Int? = null
    var gordura: Double? = null
    var foto_perfil: String? = null
    var estado: String? = null

    constructor(
        id_cliente: Int?,
        id_ginasio: Int?,
        id_plano_nutricional: Int?,
        nome: String?,
        mail: String?,
        telemovel: Int?,
        pass_salt: String?,
        pass_hash: String?,
        peso: Double?,
        altura: Int?,
        gordura: Double?,
        foto_perfil: String?,
        estado: String?
    ) {
        this.id_cliente = id_cliente
        this.id_ginasio = id_ginasio
        this.id_plano_nutricional = id_plano_nutricional
        this.nome = nome
        this.mail = mail
        this.telemovel = telemovel
        this.pass_salt = pass_salt
        this.pass_hash = pass_hash
        this.peso = peso
        this.altura = altura
        this.gordura = gordura
        this.foto_perfil = foto_perfil
        this.estado = estado
    }

    fun toJson() : JSONObject {
        val jsonObj = JSONObject()

        jsonObj.put("id_cliente", id_cliente)
        jsonObj.put("id_ginasio", id_ginasio)
        jsonObj.put("id_plano_nutricional", id_plano_nutricional)
        jsonObj.put("nome", nome)
        jsonObj.put("mail", mail)
        jsonObj.put("telemovel", telemovel)
        jsonObj.put("pass_salt", pass_salt)
        jsonObj.put("pass_hash", pass_hash)
        jsonObj.put("peso", peso)
        jsonObj.put("altura", altura)
        jsonObj.put("gordura", gordura)
        jsonObj.put("foto_perfil", foto_perfil)
        jsonObj.put("estado", estado)


        return jsonObj
    }

    companion object{
        fun fromJson(jsonObject: JSONObject) : Cliente {
            return Cliente(
                jsonObject.getInt("id_cliente"),
                jsonObject.getInt("id_ginasio"),
                jsonObject.optInt("id_plano_nutricional"),
                jsonObject.getString("nome"),
                jsonObject.getString("mail"),
                jsonObject.getInt("telemovel"),
                jsonObject.getString("pass_salt"),
                jsonObject.getString("pass_hash"),
                jsonObject.optDouble("peso"),
                jsonObject.optInt("altura"),
                jsonObject.optDouble("gordura"),
                jsonObject.getString("foto_perfil"),
                jsonObject.getString("estado")
            )
        }
    }
}
